src/ButteredBread.sol

```diff
@@ -115,14 +124,15 @@ contract ButteredBread is ERC20VotesUpgradeable, OwnableUpgradeable, IButteredBr
         _accountToLPData[_account][_lp].scalingFactor = currentScalingFactor;
 
         _mint(_account, _amount * currentScalingFactor / FIXED_POINT_PERCENT);
-        if (this.delegates(_account) == address(0)) _delegate(_account, _account);
+        _delegate(_account, BREAD.delegates(_account));
```

Comment:
```suggestion
       address delegatee = BREAD.delegates(_account);
       delegatee == address(0) ?  _delegate(_account,_account) : _delegate(_account,delegatee ) ;
```

Reply:
this snippet could also instead be made into a function that anyone can call , would reduce the amount of code and also allow a bypass for mismatched state

Reply:
seems reasonable. also can just call the function code in `_deposit` `_withdraw` instead of repeating code